### ISSUE TITLE
Check syntax of vertex and fragment shaders at same time

### ISSUE THREAD

**Issue by h4k1m0u** [KhronosGroup/glslang#2966]
Apologies in advance if the question has already been asked before (I couldn't find a similar one among the issues).

I just built glslang and I find very useful tool for someone starting up with GLSL (and OpenGL). I was wondering if it was possible, considering a GLSL vertex and fragment shader to check their syntax together? i.e. whether they would compile & link if run one after the other.

Maybe I'm understanding this wrong, but `-l` option isn't returning an error when the name of the two blocks below don't match:

In the vertex shader:

```glsl
out VS_OUT1 {
  vec2 texture_coord_vert;
  vec3 position_vert;
  vec3 normal_vert; // not used! (normal retrieved from `texture_normal`)
} vs_out;
 ```

In in the fragment shader:

```glsl
// interface block (name matches in vertex shader)                    
in VS_OUT2 {
  vec2 texture_coord_vert;
  vec3 position_vert;                                                 
  vec3 normal_vert;
} fs_in;
```

**Comment by greg-lunarg** [comment]
Yes, I believe you are correct. These should not match and an error should be given under -l.

There also does not seem to be any regression tests for this problem.
